lib/keeper/token.rb

```diff
@@ -34,26 +33,53 @@ def self.find(raw_token)
       new_token
     end
 
-    # Revokes a web token
-    def revoke
-      return if invalid?
-      Datastore.expire(claims[:jti], claims[:exp] - DateTime.now.to_i)
+    # Sets a token to the pending rotation state. The expire is set to the maxium possible time but
+    # is inherently ignored by the token's exp check and then rewritten with the revokation on
+    # rotate.
+    # @param token_jti [String] the token unique id
+    def self.rotate(token_jti)
```

Comment:
more than 1 self function, can we just make a self block with `class << self` shit is easier to read 😄